File: src/main/java/com/ocds/etender/model/dto/tender/Tender.java
package com.ocds.etender.model.dto.tender;

import com.fasterxml.jackson.annotation.JsonInclude;
import lombok.Data;

import java.util.List;

@Data
public class Tender {
    public String id;
    public String title;
    public String description;
    public String status;
    @JsonInclude(JsonInclude.Include.NON_NULL)
    public List<Item> items;

    @JsonInclude(JsonInclude.Include.NON_NULL)
    public Value minValue;//todo not find in file doc

    @JsonInclude(JsonInclude.Include.NON_NULL)
    public Value value;

    @JsonInclude(JsonInclude.Include.NON_NULL)
    public String procurementMethod;

    @JsonInclude(JsonInclude.Include.NON_NULL)
    public String procurementMethodDetails;

    @JsonInclude(JsonInclude.Include.NON_NULL)
    public String procurementMethodAdditionalInfo;//todo not find in file doc

    @JsonInclude(JsonInclude.Include.NON_NULL)
    public String procurementMethodRationale;

    @JsonInclude(JsonInclude.Include.NON_NULL)
    public String procurementCategory;

    @JsonInclude(JsonInclude.Include.NON_NULL)
    public String awardCriteria;

    @JsonInclude(JsonInclude.Include.NON_NULL)
    public String awardCriteriaDetails;

    @JsonInclude(JsonInclude.Include.NON_NULL)
    public List<String> submissionMethod;

    @JsonInclude(JsonInclude.Include.NON_NULL)
    public String submissionMethodDetails;

    @JsonInclude(JsonInclude.Include.NON_NULL)
    public Period tenderPeriod;

    @JsonInclude(JsonInclude.Include.NON_NULL)
    public Period enquiryPeriod;

    @JsonInclude(JsonInclude.Include.NON_NULL)
    public Boolean hasEnquiries;

    @JsonInclude(JsonInclude.Include.NON_NULL)
    public String eligibilityCriteria;

    @JsonInclude(JsonInclude.Include.NON_NULL)
    public Period awardPeriod;

    @JsonInclude(JsonInclude.Include.NON_NULL)
    public Period contractPeriod;//todo not find in file doc

    @JsonInclude(JsonInclude.Include.NON_NULL)
    public Integer numberOfTenderers;

    @JsonInclude(JsonInclude.Include.NON_NULL)
    public List<OrganizationReference> tenderers;//todo not find in file doc

    @JsonInclude(JsonInclude.Include.NON_NULL)
    public OrganizationReference procuringEntity;

    @JsonInclude(JsonInclude.Include.NON_NULL)
    public List<Document> documents;

    @JsonInclude(JsonInclude.Include.NON_NULL)
    public List<Milestone> milestones;//todo not find in file doc

    @JsonInclude(JsonInclude.Include.NON_NULL)
    public List<Amendment> amendments;//todo not find in file doc

    @JsonInclude(JsonInclude.Include.NON_NULL)
    public List<Lot> lots;

    @JsonInclude(JsonInclude.Include.NON_NULL)
    public LotDetails lotDetails;

    @JsonInclude(JsonInclude.Include.NON_NULL)
    public List<LotGroup> lotGroups;

    @JsonInclude(JsonInclude.Include.NON_NULL)
    public List<ParticipationFee> participationFees;//todo not find in file doc

    @JsonInclude(JsonInclude.Include.NON_NULL)
    public Criterion criteria;

    @JsonInclude(JsonInclude.Include.NON_NULL)
    public AcceleratedProcedure acceleratedProcedure;

    @JsonInclude(JsonInclude.Include.NON_NULL)
    public Classification classification;

    @JsonInclude(JsonInclude.Include.NON_NULL)
    public DesignContest designContest;

    @JsonInclude(JsonInclude.Include.NON_NULL)
    public ElectronicWorkFlows electronicWorkflows;

    @JsonInclude(JsonInclude.Include.NON_NULL)
    public JointProcurement jointProcurement;

    @JsonInclude(JsonInclude.Include.NON_NULL)
    public String legalBasis;

    @JsonInclude(JsonInclude.Include.NON_NULL)
    public Objectives objectives;

    @JsonInclude(JsonInclude.Include.NON_NULL)
    public ProcedureOutsourcing procedureOutsourcing;

    @JsonInclude(JsonInclude.Include.NON_NULL)
    public List<OrganizationReference> reviewParties;//todo not find in file doc

    @JsonInclude(JsonInclude.Include.NON_NULL)
    public Period reviewPeriod;//todo not find in file doc

    @JsonInclude(JsonInclude.Include.NON_NULL)
    public Period standstillPeriod;//todo not find in file doc

    @JsonInclude(JsonInclude.Include.NON_NULL)
    public List<String> submissionLanguages;

    @JsonInclude(JsonInclude.Include.NON_NULL)
    public List<String> submissionMethodRationale;

    @JsonInclude(JsonInclude.Include.NON_NULL)
    public DynamicPurchasingSystem dynamicPurchasingSystem;

    @JsonInclude(JsonInclude.Include.NON_NULL)
    public Framework framework;

    @JsonInclude(JsonInclude.Include.NON_NULL)
    public Boolean requiresElectronicCatalogue;//todo not find in file doc
}

File: src/main/kotlin/com/procurement/access/model/dto/ocds/Tender.kt
package com.procurement.access.model.dto.ocds

import com.fasterxml.jackson.annotation.JsonCreator
import com.fasterxml.jackson.annotation.JsonInclude
import com.fasterxml.jackson.annotation.JsonProperty
import com.fasterxml.jackson.databind.annotation.JsonDeserialize
import com.procurement.access.domain.model.enums.AwardCriteria
import com.procurement.access.domain.model.enums.AwardCriteriaDetails
import com.procurement.access.domain.model.enums.ExtendedProcurementCategory
import com.procurement.access.domain.model.enums.LegalBasis
import com.procurement.access.domain.model.enums.MainProcurementCategory
import com.procurement.access.domain.model.enums.ProcurementMethod
import com.procurement.access.domain.model.enums.ProcurementMethodModalities
import com.procurement.access.domain.model.enums.SubmissionLanguage
import com.procurement.access.domain.model.enums.SubmissionMethod
import com.procurement.access.domain.model.enums.TenderStatus
import com.procurement.access.domain.model.enums.TenderStatusDetails
import com.procurement.access.infrastructure.entity.CNEntity
import com.procurement.access.model.dto.databinding.BooleansDeserializer

@JsonInclude(JsonInclude.Include.NON_NULL)
data class Tender @JsonCreator constructor(

        val id: String?,

        var title: String,

        var description: String,

        var status: TenderStatus,

        var statusDetails: TenderStatusDetails,

        var classification: Classification,

        val mainProcurementCategory: MainProcurementCategory,

        var additionalProcurementCategories: List<ExtendedProcurementCategory>?,

        val procurementMethod: ProcurementMethod,

        val procurementMethodDetails: String,

        var procurementMethodRationale: String?,

        var procurementMethodAdditionalInfo: String?,

        val submissionMethod: List<SubmissionMethod>,

        val submissionMethodDetails: String,

        val submissionMethodRationale: List<String>,

        val submissionLanguages: List<SubmissionLanguage>?,

        val eligibilityCriteria: String,

        val acceleratedProcedure: AcceleratedProcedure,

        val designContest: DesignContest,

        val electronicWorkflows: ElectronicWorkflows,

        val jointProcurement: JointProcurement,

        val procedureOutsourcing: ProcedureOutsourcing,

        val framework: Framework,

        val dynamicPurchasingSystem: DynamicPurchasingSystem,

        val legalBasis: LegalBasis,

        @JsonInclude(JsonInclude.Include.NON_NULL)
        val procuringEntity: OrganizationReference?,

        var awardCriteria: AwardCriteria?,

        @JsonInclude(JsonInclude.Include.NON_NULL)
        @field:JsonProperty("awardCriteriaDetails") @param:JsonProperty("awardCriteriaDetails") val awardCriteriaDetails: AwardCriteriaDetails?,

        @JsonInclude(JsonInclude.Include.NON_EMPTY)
        @field:JsonProperty("criteria") @param:JsonProperty("criteria") val criteria: List<CNEntity.Tender.Criteria> = emptyList(),

        @JsonInclude(JsonInclude.Include.NON_EMPTY)
        @field:JsonProperty("conversions") @param:JsonProperty("conversions") val conversions: List<CNEntity.Tender.Conversion> = emptyList(),

        @get:JsonProperty("requiresElectronicCatalogue")
        @field:JsonDeserialize(using = BooleansDeserializer::class)
        val requiresElectronicCatalogue: Boolean,

        var contractPeriod: ContractPeriod?,

        var tenderPeriod: Period? = null,

        var enquiryPeriod: Period? = null,

        var value: Value,

        val lotGroups: List<LotGroup>,

        var lots: List<Lot>,

        var items: List<Item>,

        var documents: List<Document>?,

        var procurementMethodModalities: Set<ProcurementMethodModalities>?,

        var electronicAuctions: ElectronicAuctions?,

        @JsonInclude(JsonInclude.Include.NON_NULL)
        @field:JsonProperty("otherCriteria") @param:JsonProperty("otherCriteria") val otherCriteria: OtherCriteria? = null
)
